src/Plot.js

```diff
@@ -138,19 +144,28 @@ export default class Plot extends Viz {
       this._sizeScaleD3 = () => this._sizeMin;
     }
 
+    const x2Exists = data.some(d => d.x2),
+          y2Exists = data.some(d => d.y2);
```

Comment:
Holy shit I didn't know about `Array.some`, super cool!

There's an interesting edge-case we need to catch here, theoretically a zero value will return falsey (but might actually be what a user wants), so we can't just check for `d.x2`, we would need to catch for `d.x2 !== undefined`.